### PR TITLE
Document ensure_open_zip

### LINKED ISSUE #26: Ensure zip file?
Hello,
I love the project. However I am missing something like an `ensure_zip_file` function. While ensuring that the actual `.zip` is there can be done with `ensure`, it would be nice to have a functionality, where I ensure a file from this zip is there, and load this file as fileobject, to then read it in however it is needed.
Poking around in the code I found `Module.ensure_open_zip`. I think this contextmanager can be easily wrapped for the API, to get what I need like this:

```python
def ensure_open_zip(
    key: str,
    *subkeys: str,
    url: str,
    inner_path: str,
    name: Optional[str] = None,
    download_kwargs: Optional[Mapping[str, Any]] = None,
):
    _module = Module.from_key(key, ensure_exists=True)
    return _module.ensure_open_zip(*subkeys, url=url, inner_path=inner_path, name=name)

```

If it's fine with you I could make a PR and add it (with tests and doc of course)?

### PR DESCRIPTION
**Pull request by dobraczka**
Closes #26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,20 @@ rdf_graph: rdflib.Graph = pystow.ensure_rdf('rhea', url=url)
...
 
+If you're data comes with a lot of different files in an archive you can ensure the archive is downloaded and get specific files from it:
+
+```python
+import numpy as np
+from pystow import module
+url = "/my/url/with/data.zip"
+# the path inside the archive to the file you want
+inner_path = "root_folder/embeddings.npy"
+with module("test").ensure_open_zip(url=url, inner_path=inner_path) as file:
```
Comment (cthoyt) [inline README.md:116]: would be better to finally expose this as a top-level method, similarly to the other ones

File: README.md [hunk 2]
```diff
@@ -105,6 +105,20 @@ rdf_graph: rdflib.Graph = pystow.ensure_rdf('rhea', url=url)
 
 Also see `pystow.ensure_excel()`, `pystow.ensure_rdf()`, `pystow.ensure_zip_df()`, and `pystow.ensure_tar_df()`.
 
+If you're data comes with a lot of different files in an archive you can ensure the archive is downloaded and get specific files from it:
+
+```python
+import numpy as np
+from pystow import module
+url = "/my/url/with/data.zip"
```
Comment (cthoyt) [inline README.md:113]: please make a working example

**Comment by cthoyt** [comment]
@dobraczka can you resolve the conflicts with the main branch please? I can take care of adding the top-level functionality after.

**Comment by cthoyt** [comment]
Thanks @dobraczka! I've made the updates and merged this. I'll add a few new pull requests with new functionality based on this, too, then make a release :)

**Comment by dobraczka** [comment]
Thank you so much for adding this new functionality 🙇🏼

**Pull request merged**